packages/@aws-cdk/aws-apigateway/lib/gateway-response.ts

```diff
@@ -0,0 +1,186 @@
+import { Construct, Resource } from '@aws-cdk/core';
+import { CfnGatewayResponse } from './apigateway.generated';
+import { IRestApi } from './restapi';
+
+/**
+ * Properties for a new gateway response.
+ */
+export interface GatewayResponseProps extends GatewayResponseOptions {
+  /**
+   * Rest api resource to target.
+   */
+  readonly restApi: IRestApi;
+
+  /**
+   * Name of the gateway response.
+   * @default - Auto generated
+   */
+  readonly gatewayResponseName?: string;
```

Comment:
There's no gatewayResponseName property - https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigateway-gatewayresponse.html#aws-resource-apigateway-gatewayresponse-properties

Drop this.

Reply:
yes its not there but added it because i was getting below lint error :
 "Every Resource must have a single physical name construction property, " +
    "with a name that is an ending substring of <cfnResource>Name"